jobwatcher/jobwatcher.py

```diff
@@ -51,17 +52,56 @@ def get_asg_name(stack_name, region, proxy_config):
 
     return asg_name
 
-def get_instance_properties(instance_type):
+
+def read_cfnconfig():
+    cfnconfig_params = {}
+    with open(cfnconfig_file) as f:
+        for kvp in f:
+            key, value = kvp.partition('=')[::2]
+            cfnconfig_params[key.strip()] = value.strip()
+    return cfnconfig_params
+
+
+def get_vcpus_from_pricing_file(instance_type):
     with open(pricing_file) as f:
         instances = json.load(f)
         try:
-            slots = int(instances[instance_type]["vcpus"])
-            log.info("Instance %s has %s slots." % (instance_type, slots))
-            return {'slots': slots}
+            vcpus = int(instances[instance_type]["vcpus"])
+            log.info("Instance %s has %s vcpus." % (instance_type, vcpus))
+            return vcpus
         except KeyError as e:
             log.error("Instance %s not found in file %s." % (instance_type, pricing_file))
             exit(1)
 
+
+def get_instance_properties(instance_type):
+    cfnconfig_params = read_cfnconfig()
+    try:
+        cfn_scheduler_slots = cfnconfig_params["cfn_scheduler_slots"]
+        slots = 0
+        if cfn_scheduler_slots.isdigit():
```

Comment:
I'd restructure this closer to a `switch` statement...
```
if cfn_scheduler_slots == ''vcpus":
...
elif cfn_scheduler_slots == ''vcpus":
...
else: # it's a digit or throw an error
```
Having nested if blocks makes it confusing to read.

Reply:
The problem with that is that we would either have 2 calls to get_vcpus_from_pricing_file or an unnecessary call in case it is a digit.

Reply:
On second thought, maybe code clarity is better than making an excess call to get_vcpus_from_pricing_file

Reply:
Done. This will be done in the next PR